### ISSUE TITLE
Check Talsim DB version when trying to open it

### ISSUE THREAD

**Issue by jamaa** [sydroconsult/QTalsim#4]
When trying to connect to a Talsim DB that does not already have the `Geometry` field, the following error occurs:

```
2024-03-25T08:56:53     WARNING    Traceback (most recent call last):
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim_sqllite_dialog.py", line 122, in createSubBasinsLayerInputDB
              self.createSystemElementsAndOutflows()
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim_sqllite_dialog.py", line 225, in createSystemElementsAndOutflows
              self.addPolygonsSubBasins()
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim_sqllite_dialog.py", line 247, in addPolygonsSubBasins
              if str(feature[self.geometryFieldName]).strip().upper() != 'NULL' and feature['ElementTypeCharacter'] == self.elementTypeCharacter:
             KeyError: 'Geometry'
```

We have two options:
* Create the missing field automatically
* Display an error message to the user saying that this version of the database is not supported

**Comment by jamaa** [comment]
In future, Talsim5 will add this field itself, but in case someone tries to connect an old database, an error should be displayed.

The database contains a table `__EFMigrationsHistory` which can be used to determine the "version" of the database. QTalsim should check this table and display an error if the version is too old, and a warning if the version is newer than expected.

To retrieve the current database version:
```sql
SELECT MigrationId FROM __EFMigrationsHistory ORDER BY MigrationId DESC LIMIT 1;
```
Then compare the datestring at the beginning of that field's value with the expected version number.

I will let you know in which version the Geometry field was added.

**Comment by EmaDozer** [comment]
With the '20240325161029_add_Geometry_SystemElement' migration the new Geometry field was added to the database.

**Comment by lorenz3tla** [comment]
Implemented with v1.3.0: Error message is displayed if the version of Talsim DB is older than '20240325161029_add_Geometry_SystemElement'. Warning is displayed if the version is newer than '20240325161029_add_Geometry_SystemElement'.
